src/core/pal/pointset.cpp

```diff
@@ -135,15 +135,17 @@ void PointSet::createGeosGeom() const
   GEOSCoordSequence *coord = GEOSCoordSeq_create_r( geosctxt, nbPoints + ( needClose ? 1 : 0 ), 2 );
   for ( int i = 0; i < nbPoints; ++i )
   {
-    GEOSCoordSeq_setX_r( geosctxt, coord, i, x[i] );
-    GEOSCoordSeq_setY_r( geosctxt, coord, i, y[i] );
+    // GEOS doesn't supports nan coordinates
+    GEOSCoordSeq_setX_r( geosctxt, coord, i, std::isnan( x[i] ) ? 0 : x[i] );
```

Comment:
Shouldn't we just skip these? (Although I doubt a nan vertex will reach this far anyway)

Reply:
Done